### PR TITLE
Region Flag 

### PR DESCRIPTION
**Pull request by riteshsonawane1372**
## Fixes Issue

#12 
`` awsctl get ec2 --region us-east-2``

### CODE REVIEW

**Comment by riteshsonawane1372** [comment]
Should I create a list of regions and compare it with the input of --region?

**Comment by surajnarwade** [comment]
@riteshsonawane1372 currently this flag is only for ec2 command, can we make it global flag and make it work for all the current and future commands?

similar to how --kubeconfig in kubectl

File: cmd/root.go [hunk 1]
```diff
@@ -31,13 +25,6 @@ func Execute() {
...
-	// Here you will define your flags and configuration settings.
-	// Cobra supports persistent flags, which, if defined here,
-	// will be global for your application.
-
-	// rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.awsctl.yaml)")
-
-	// Cobra also supports local flags, which will only run
-	// when this action is called directly.
 	rootCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+	rootCmd.PersistentFlags().String("region", "", "--region=us-east-1")
```
Comment (surajnarwade) [inline cmd/root.go:29]: Write help message

File: pkg/ec2/getEc2Info.go [hunk 2]
```diff
@@ -58,12 +59,23 @@ func DescribeInstance(tagName string, tagValue []string) []EC2Instance {
...
 	ctx := context.TODO()
 	cfg, err := config.LoadDefaultConfig(ctx)
 	if err != nil {
 		log.Fatal("Config Error Occured", err)
 	}
+
+	region := cmd.Root().PersistentFlags().Lookup("region").Value.String()
+
+	if region != "" {
+		opts := func(o *ec2.Options) {
```
Comment (surajnarwade) [inline pkg/ec2/getEc2Info.go:73]: can you explain this function? and also can you add comments wherever necessary in the code.

Comment (riteshsonawane1372) [reply pkg/ec2/getEc2Info.go:73]: ``ec2.NewfromConfig``  takes options in the form of arrays of functions https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2#Options

File: cmd/root.go (hunk 1 above)
Comment (riteshsonawane1372) [reply cmd/root.go:29]: Something like ``awsctl get ec2 --region --help`` ??

Comment (surajnarwade) [reply cmd/root.go:29]: instead of `--region=us-east-1`, add something meaningful, maybe you can refer AWS CLI

**Review by surajnarwade — APPROVED** [review]

**Comment by riteshsonawane1372** [comment]
Should I make changes to other services too?  @surajnarwade

**Comment by surajnarwade** [comment]
can we make this somehow default that any new command will automatically will have this?

**Comment by riteshsonawane1372** [comment]
> can we make this somehow default that any new command will automatically will have this?

I can do one thing, change env variable bcoz aws give more priority to the env variable than the config file. Any other option you can recommend?

**Comment by surajnarwade** [comment]
@riteshsonawane1372 yes go for it

**Comment by riteshsonawane1372** [comment]
I have tested it and everything works good.

**Pull request merged**